File: sdk/go/owners.go
package marmot

import (
	"context"

	apiclient "github.com/marmotdata/marmot/sdk/go/internal/gen/client"
	"github.com/marmotdata/marmot/sdk/go/internal/gen/client/owners"
	"github.com/marmotdata/marmot/sdk/go/internal/gen/models"
)

// OwnerSearchResults is the response from OwnersService.Search.
type OwnerSearchResults = models.V1TeamsSearchOwnersResponse

// OwnerSearchOptions filters OwnersService.Search.
type OwnerSearchOptions struct {
	Limit int64
}

// OwnersService searches the catalog for asset owners (users and teams).
type OwnersService struct {
	gen *apiclient.Marmot
}

// Search returns owners matching q.
func (s *OwnersService) Search(ctx context.Context, q string, opts OwnerSearchOptions) (*OwnerSearchResults, error) {
	p := owners.NewGetOwnersSearchParams().WithContext(ctx).WithQ(q)
	if opts.Limit > 0 {
		p = p.WithLimit(&opts.Limit)
	}
	resp, err := s.gen.Owners.GetOwnersSearch(p)
	if err != nil {
		return nil, mapErr(err)
	}
	return resp.Payload, nil
}

File: sdk/python/src/marmot/resources/owners.py
"""Search the catalog for asset owners (users and teams)."""

from __future__ import annotations

from typing import cast

from marmot._adapter import unwrap
from marmot._gen.api.owners import get_owners_search
from marmot._gen.client import AuthenticatedClient
from marmot._gen.models.search_owners_response import SearchOwnersResponse
from marmot._gen.types import UNSET, Unset


class OwnersResource:
    def __init__(self, client: AuthenticatedClient) -> None:
        self._c = client

    def search(self, query: str, *, limit: int | None = None) -> SearchOwnersResponse:
        """Return owners matching the query."""
        limit_arg: int | Unset = limit if limit is not None else UNSET
        return cast(
            SearchOwnersResponse,
            unwrap(get_owners_search.sync_detailed(client=self._c, q=query, limit=limit_arg)),
        )
